src/main/java/de/tum/in/www1/artemis/web/rest/ModelingSubmissionResource.java

```diff
@@ -191,10 +192,23 @@ public ModelingSubmissionResource(SubmissionRepository submissionRepository, Res
         var gradingCriteria = gradingCriterionService.findByExerciseIdWithEagerGradingCriteria(modelingExercise.getId());
         modelingExercise.setGradingCriteria(gradingCriteria);
         final User user = userRepository.getUserWithGroupsAndAuthorities();
-        if (!authCheckService.isAtLeastTeachingAssistantForExercise(modelingExercise, user)) {
+        if (!authCheckService.isAtLeastTeachingAssistantForExercise(modelingExercise, user)
+                || (resultId > 0 && !authCheckService.isAtLeastInstructorForExercise(modelingExercise, user))) {
             return forbidden();
         }
-        modelingSubmission = modelingSubmissionService.lockAndGetModelingSubmission(submissionId, modelingExercise, correctionRound);
+
+        if (resultId > 0) {
```

Comment:
We now have quite some duplicated code even if the code is not specific to modeling / file upload / text.
Would it not make sense to extract it into a common method?

Reply:
I looked into the code, for these calls in general, extraction and unifications across the different exercise types could be done. I guess this should be done carefully and well tested. I think a follow-up would be a good idea for that.